hack-sound-server

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python3
+
+import gi
+import logging
+import json
+import os
+import signal
+import uuid
+gi.require_version('GLib', '2.0')  # noqa
+gi.require_version('Gst', '1.0')   # noqa
+gi.require_version('Json', '1.0')  # noqa
+from gi.repository import Gio
+from gi.repository import GLib
+from gi.repository import GObject
+from gi.repository import Gst
+from gi.repository import Json
+
+
+_logger = logging.getLogger(__name__)
+
+
+def get_datadir():
+    return os.path.join(GLib.get_user_data_dir(), "hack-sound-server")
+
+def get_metadata_path():
+    return os.path.join(get_datadir(), "metadata.json")
+
+def get_sounds_dir():
+    return os.path.join(get_datadir(), "sounds")
+
+
+class HackSoundPlayer(GObject.Object):
+    __gsignals__ = {
+        'eos': (GObject.SIGNAL_RUN_FIRST, None, ()),
+        'error': (GObject.SIGNAL_RUN_FIRST, None, (object, str))
+    }
+
+    def __init__(self, metadata, sender):
+        GObject.Object.__init__(self)
+        self.metadata = metadata
+        self.sender = sender
+        self.pipeline = self._build_pipeline()
+        bus = self.pipeline.get_bus()
+        bus.add_signal_watch()
+        bus.connect("message", self.__bus_message_cb)
+
+    def play(self):
+        self.pipeline.set_state(Gst.State.PLAYING)
+
+    @property
+    def sound_location(self):
+        return os.path.join(get_sounds_dir(), self.metadata["sound-file"])
+
+    def _build_pipeline(self):
+        spipeline = "filesrc location=\"%s\" ! decodebin ! autoaudiosink" % self.sound_location
+        return Gst.parse_launch(spipeline)
+
+    def __bus_message_cb(self, unused_bus, message):
+        if message.type == Gst.MessageType.EOS:
+            self.emit("eos")
+        elif message.type == Gst.MessageType.ERROR:
+            error, debug = message.parse_error()
+            self.emit("error", error, debug)
+
+class HackSoundServer(Gio.Application):
+
+    _DBUS_NAME = "com.endlessm.HackSoundServer"
+    _DBUS_UNKNOWN_SOUND_EVENT_ID = "com.endlessm.HackSoundServer.UnknownSoundEventID"
+    _DBUS_GST_ERROR = "com.endlessm.HackSoundServer.GStreamerError"
+    _DBUS_XML = """
+    <node>
+      <interface name='com.endlessm.HackSoundServer'>
+        <method name='PlaySound'>
+          <arg type='s' name='sound_event' direction='in'/>
+          <arg type='s' name='uuid' direction='out'/>
+        </method>
+        <signal name='Error'>
+          <arg type='s' name='uuid'/>
+          <arg type='s' name='error_message'/>
+          <arg type='s' name='error_domain'/>
+          <arg type='i' name='error_code'/>
+          <arg type='s' name='debug'/>
+        </signal>
+      </interface>
+    </node>
+    """
+
+    def __init__(self, metadata):
+        super().__init__(application_id=self._DBUS_NAME,
+                         flags=Gio.ApplicationFlags.IS_SERVICE)
+        self._dbus_id = None
+        self.metadata = metadata
+        self.hold()
```

Comment:
I'm not 100% convinced that we need to keep this server running all the time. Is the playback latency a concern if we don't?